### PR TITLE
Add additonal loot to bank vaults

### PR DESCRIPTION
**Pull request by Firestorm01X2**
Add additonal loot to bank vaults

#### Summary
SUMMARY: Balance "Add additonal loot to bank vaults"

#### Purpose of change
There was PR: https://github.com/CleverRaven/Cataclysm-DDA/pull/32634
that significantly reduce loot quality of banks vaults. This PR has its points like removing clearly lab stuff or power armor but banks are guarded and protected so player should get noticeable reward for getting there.
It is better to add some additional to bank vaults to make raiding banks  valuable.

#### Describe the solution
Now bank vaults additonally has good chance of containing.
* Some rare and useful CBM. But without extremes. No weaponized CBMs. But still there are still things like teleportation CBM. It was present before my change, so I am keeping it.
* Imported, rare or otherwise very obscure guns. AKs, FAMAS e. t. c.
* Rare antique medieval melee weapons (without fakes) and antique guns.
* Pricey and somewhat rare military guns including Rivtech guns. Good chance of appearing sniper rifle. Without heavy weapons or LMGs.
* Rare recipes and reports from labs.
* Increased chance to spawn schematics.

Technically all stuff above considered pricey a or rare and technically could present in banks (in game world). Well, maybe not in that proportions and probably not military rifles, but it is compromise to make banks raiding more attractive and worth time considering guard bots, locked doors and safes.
Still no lab stuff (with exception of some high tech CBMs that was present before my change and recipes).

#### Describe alternatives you've considered
Make banks already looted after many months of marauding. Remove bots (or leave their destroyed bodies), make safes violently open etc.

#### Testing

1) Observe bank vaults. It should contian more CBMs and rare and obscure weapons.

#### Additional context
Save for testing.
[banks2.zip](https://github.com/cataclysmbnteam/Cataclysm-BN/files/7864700/banks2.zip)

Character already equipped with jackhammer and clairvoyance artifact. Road map already read.
Banks to the city on the West. Looks for "$" signs on map.

### CODE REVIEW

**Comment by jrickter88** [comment]
"that significantly reduce loot quality of banks vaults."

As if they werent bad enough.

**Comment by Coolthulhu** [comment]
The new groups have high weights, so make other things less likely. Rare recipes are better than most of the new groups.
For example, fake medieval weapons are disappointing garbage and not really fitting.

Would be better if you split the group into two subgroups, essentially "bank stuff that's good" (recipe books, good bionics) and "bank garbage" like money, gold, holy symbols etc.
Each group would have its own chance to be picked, so that bloating one of them wouldn't affect the chances of the other one.

**Comment by Night-Pryanik** [comment]
> Describe alternatives you've considered

Make banks already looted after many months of marauding. Remove bots (or leave their destroyed bodies), make safes violently open etc.

**Comment by chaosvolt** [comment]
That's a good suggestion, but unrealistic to have every bank vault be cracked open and looted when every house and store is still usually full, and clashes with the supposed suddenness of the cataclysm.

Plus this PR at least makes it internally consistent, last I checked in DDA bank vaults are still completely sealed up and secure despite also being basically empty, which is completely ass-backwards as far as game balance goes.

**Comment by Night-Pryanik** [comment]
> supposed suddenness of the cataclysm.

I'm not sure where did this idea came from as old lore (which AFAIK BN is based on) clearly states that there were many months of civil disorders and marauding which undoubtedly led to looted banks. Assuming this concept is right, intact shops and houses is a bug that should be fixed, not a normal thing.

> DDA bank vaults are still completely sealed up and secure despite also being basically empty

What do you mean by "basically empty"? They are full of stuff that _should_ be in IRL bank vaults, precisely because they are sealed up.

**Comment by chaosvolt** [comment]
The objective here seems to be giving bank vaults a reason to be worth breaking into, which Firestorm aims to accomplish without adding too much wacky military stuff, instead favoring corporate science tech that would logically be in secure storage (and consistent with BN retaining a sci-fi setting). Coolthulhu's proposal of splitting things up into "good loot" and "bank filler" would be a good way for us to fine-tune the spawns a bit as well, since that gives us more control over how much corporate sci-fi stuff will show up without messing up how much mundane bank stuff spawns.

> What do you mean by "basically empty"? They are full of stuff that should be in IRL bank vaults, precisely because they are sealed up.

They are at least empty of anything worth bothering with looting them, which makes them meaningless flavor locations. Worse, they're meaningless flavor locations that actively drain the player's resources before they reveal how meaningless they are.

Plus again, if the cataclysm gave plenty of time for things to get looted, then that is inconsistent with banks spawning sealed up in the first place. It feels a bit disingenuous to me that this suggestion only now seems to be coming up as a talking point.

**Comment by Night-Pryanik** [comment]
> The objective here seems to be giving bank vaults a reason to be worth breaking into, which Firestorm aims to accomplish

As if I didn't understand that. Actually it was me who recommended Firestorm to open this particular PR with restoration of old loot list for banks.

> the cataclysm gave plenty of time for things to get looted, then that is inconsistent with banks spawning sealed up in the first place.

In the new DDA lore the riots began only a month before the Cataclysm. By this new lore there wasn't enough time to loot most locations, banks included.

File: data/json/itemgroups/Locations_MapExtras/locations.json [hunk 1]
```diff
@@ -2054,18 +2054,33 @@
...
     "items": [
-      { "group": "ammo_atomic_batteries", "prob": 10 },
+      { "group": "vault_good", "prob": 500 },
+      { "group": "vault_bad", "prob": 500 }
+    ]
+  },
+  {
+    "type": "item_group",
+    "id": "vault_good",
+    "//": "Usefull or good stuff for bank vaults.",
```
Comment (Night-Pryanik) [inline data/json/itemgroups/Locations_MapExtras/locations.json:2064]:
```suggestion
    "//": "Useful or good stuff for bank vaults.",
```

File: data/json/itemgroups/Locations_MapExtras/locations.json [hunk 2]
```diff
@@ -2099,10 +2117,10 @@
       [ "coin_quarter", 2 ],
       [ "straw_doll", 2 ],
       [ "petrified_eye", 1 ],
-      [ "spiral_stone", 1 ],
       [ "etched_skull", 1 ]
     ]
   },
+
```
Comment (chaosvolt) [inline data/json/itemgroups/Locations_MapExtras/locations.json:2123]: Lint issue likely caused by the added newline here.

**Comment by chaosvolt** [comment]
Favoring obscure collectors firearms may be a useful angle to go for.

**Comment by Firestorm01X2** [comment]
Added Rare recipes, reorganized groups.
Ready for review.

**Comment by Coolthulhu** [comment]
Looks much better now. Just the few boring CBMs should be removed and it'll be good to go.

Ideally, drops of worse stuff, like antiques and bad, would we sampled more times (so you get say, 3 different melee antiques and 10 rolls of garbage, or 1 CBM + 1 book), but that's just details. If you can easily put it in, it would be nice, but it's OK without it.

**Comment by Firestorm01X2** [comment]
> Looks much better now. Just the few boring CBMs should be removed and it'll be good to go.
> 
> Ideally, drops of worse stuff, like antiques and bad, would we sampled more times (so you get say, 3 different melee antiques and 10 rolls of garbage, or 1 CBM + 1 book), but that's just details. If you can easily put it in, it would be nice, but it's OK without it.

I recommend to give to try.
Technically antiques are not necessary that bad. It contains a lot good melee options.

**Pull request merged**